boards.txt

```diff
@@ -33287,6 +33287,266 @@ deneyapkart.menu.EraseFlash.all.upload.erase_cmd=-e
 
 ##############################################################
 
+deneyapkartv2.name=Deneyap Kart v2
+
+deneyapkartv2.vid.0=0x303a
+deneyapkartv2.pid.0=0x82EB
+
+deneyapkartv2.bootloader.tool=esptool_py
+deneyapkartv2.bootloader.tool.default=esptool_py
+
+deneyapkartv2.upload.tool=esptool_py
+deneyapkartv2.upload.tool.default=esptool_py
+deneyapkartv2.upload.tool.network=esp_ota
+
+deneyapkartv2.upload.maximum_size=1310720
+deneyapkartv2.upload.maximum_data_size=327680
+deneyapkartv2.upload.flags=
+deneyapkartv2.upload.extra_flags=
+deneyapkartv2.upload.use_1200bps_touch=false
+deneyapkartv2.upload.wait_for_upload_port=false
+
+deneyapkartv2.serial.disableDTR=false
+deneyapkartv2.serial.disableRTS=false
+
+deneyapkartv2.build.tarch=xtensa
+deneyapkartv2.build.bootloader_addr=0x0
+deneyapkartv2.build.target=esp32s3
+deneyapkartv2.build.mcu=esp32s3
+deneyapkartv2.build.core=esp32
+deneyapkartv2.build.variant=deneyapkartv2
+deneyapkartv2.build.board=DYDKV2
+
+deneyapkartv2.build.usb_mode=1
+deneyapkartv2.build.cdc_on_boot=1
+deneyapkartv2.build.msc_on_boot=0
+deneyapkartv2.build.dfu_on_boot=0
+deneyapkartv2.build.f_cpu=240000000L
+deneyapkartv2.build.flash_size=4MB
+deneyapkartv2.build.flash_freq=80m
+deneyapkartv2.build.flash_mode=dio
+deneyapkartv2.build.boot=qio
+deneyapkartv2.build.boot_freq=80m
+deneyapkartv2.build.partitions=default
+deneyapkartv2.build.defines=-DBOARD_HAS_PSRAM
+deneyapkartv2.build.loop_core=
+deneyapkartv2.build.event_core=
+deneyapkartv2.build.psram_type=opi
+deneyapkartv2.build.memory_type={build.boot}_{build.psram_type}
+
+## IDE 2.0 Seems to not update the value
+deneyapkartv2.menu.JTAGAdapter.default=Disabled
+deneyapkartv2.menu.JTAGAdapter.default.build.copy_jtag_files=0
+deneyapkartv2.menu.JTAGAdapter.builtin=Integrated USB JTAG
+deneyapkartv2.menu.JTAGAdapter.builtin.build.openocdscript=esp32s3-builtin.cfg
+deneyapkartv2.menu.JTAGAdapter.builtin.build.copy_jtag_files=1
+deneyapkartv2.menu.JTAGAdapter.external=FTDI Adapter
+deneyapkartv2.menu.JTAGAdapter.external.build.openocdscript=esp32s3-ftdi.cfg
+deneyapkartv2.menu.JTAGAdapter.external.build.copy_jtag_files=1
+deneyapkartv2.menu.JTAGAdapter.bridge=ESP USB Bridge
+deneyapkartv2.menu.JTAGAdapter.bridge.build.openocdscript=esp32s3-bridge.cfg
+deneyapkartv2.menu.JTAGAdapter.bridge.build.copy_jtag_files=1
+
+deneyapkartv2.menu.PSRAM.opi=OPI PSRAM
+deneyapkartv2.menu.PSRAM.opi.build.defines=-DBOARD_HAS_PSRAM
+deneyapkartv2.menu.PSRAM.opi.build.psram_type=opi
+deneyapkartv2.menu.PSRAM.disabled=Disabled
+deneyapkartv2.menu.PSRAM.disabled.build.defines=
+deneyapkartv2.menu.PSRAM.disabled.build.psram_type=qspi
+deneyapkartv2.menu.PSRAM.enabled=QSPI PSRAM
+deneyapkartv2.menu.PSRAM.enabled.build.defines=-DBOARD_HAS_PSRAM
+deneyapkartv2.menu.PSRAM.enabled.build.psram_type=qspi
```

Comment:
If you are using same module on all devkits, the PSRAM can be only OPI or only QSPI. Cannot work in both modes. Please remove the "unsupported" one.